core/src/main/scala/org/http4s/QValue.scala

```diff
@@ -98,10 +101,29 @@ object QValue {
   def unsafeFromString(s: String): QValue =
     fromString(s).fold(throw _, identity)
 
+  private[http4s] val parser: Parser[QValue] = {
+    import cats.parse.Parser.{char => ch, _}
+    import cats.parse.Rfc5234._
+
+    val optWS = (crlf.rep.with1 ~ wsp.rep1).rep
+
+    val qValue = string1(ch('0') ~ (ch('.') ~ digit.rep1).rep)
+      .map(
+        QValue
+          .fromString(_)
```

Comment:
Wow.  Our `.fromString` is laxer than the spec: a qValue can only have a precision of three digits.  I guess that laxity probably isn't hurting anyone?

Reply:
Yeah, I have not touched `def fromString(s: String): ParseResult[QValue]` since it is public API. Not sure if it is still makes sense

Reply:
Let's leave it for this PR and we'll consider what we want in 1.0.

Reply:
Ok